src/main/scala/firrtl/util/BackendCompilationUtilities.scala

```diff
@@ -176,7 +187,11 @@ object BackendCompilationUtilities extends LazyLogging {
           triggered = triggered || (assertionMessageSupplied && line.contains(assertionMsg))
           logger.info(line)
         },
-        logger.warn(_)
+        stderrLine => {
+          triggered = triggered || stderrLine
+            .contains("Assertion failed") || (assertionMessageSupplied && stderrLine.contains(assertionMsg))
+          logger.warn(stderrLine)
+        }
       )
     // Fail if a line contained an assertion or if we get a non-zero exit code
     //  or, we get a SIGABRT (assertion failure) and we didn't provide a specific assertion message
```

Comment:
Rather than hardcoding the assertion message, can you instead make this:
1. do the same that the stdout check does and allow the user to specify the assertion message
2. Make the return value below do the right thing even without providing Assertion failed? it's already looking for non-zero exit codes, are MFC tests that have assertion failures not returning non-zero?